contrib/internal/httputil/trace_test.go

```diff
@@ -182,3 +182,38 @@ func TestTraceAndServe(t *testing.T) {
 		assert.Equal("200", spans[0].Tag(ext.HTTPCode))
 	})
 }
+
+func TestTraceAndServeHost(t *testing.T) {
+
+	handler := func(w http.ResponseWriter, r *http.Request) {
```

Comment:
```suggestion
	handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```
I think this could reduce a bit of duplication with the casting below.